design-proposals/edge-infra-cve-tracking.md

```diff
@@ -0,0 +1,184 @@
+  CVEs on the given version. The Release Notes should be published in a
+  structured way, so that it can be parsed and used by the orchestrator. For
+example Google Container-Optimized OS published the list of Fixed CVEs for each
+versions on every release, for example
+https://cloud.google.com/container-optimized-os/docs/release-notes/m121. OS
+Resource Manager (OSRM) can populate information about the fixed CVEs when
+creating the OSProfile. For existing CVEs instead it must periodically query the
+Release Service to get the published file, and fill those information into the
+OSProfile.
+
+For Ubuntu, instead, there are multiple ways to get these information:
+- [Ubuntu CVE Tracker](https://ubuntu.com/security/cves): this is a web
+  interface to navigate CVEs, you can filter by Ubuntu versions, and check the
+  status of the CVE.
+- [CVE git repo](https://git.launchpad.net/ubuntu-cve-tracker/) Canonical
+  maintain a public git repo with structured CVE data
+- [Ubuntu OVAL](https://ubuntu.com/security/oval) Ubuntu provides vulnerability
+  feeds as structured XML file. This contains all CVEs affecting Ubuntu
+  releases, both for packages provided in the Image and on other OS components.
+- [osv.dev](https://osv.dev) Ubuntu participates in OSV.dev, that provides a
+  standardized CVE APIs. However, this is able to provide CVE for packages, not
+  general CVEs affecting a certain Ubuntu release, for example affecting the
+  kernel.
+
+The best choice needs to be agreed with a PoCs, this should cover both
+information about existing CVEs and fixed CVEs on a given Ubuntu version. The
+work of fetching CVEs for Ubuntu should be done by OSRM, that should fill the
+fixed CVEs at OSProfile creation time, and periodically update the existing
+CVEs.
+
+### Instance
+In the case of Instance CVEs, the information regards the software that runs at
+runtime in the Edge Node. For EMT, the CVEs of the runtime software are
+effectively the same that affect the linked OS Profile. The `cves` field can
+thus be filled automatically with information from the OS Profile for an initial
+implementation, while we implement the reporting mechanism on the Edge Node
+software. This can be done by Maintenance Manager (MM), when the Platform Update
+Agent (PUA) sends its `PlatformUpdateRequest`. At that moment, MM, could fetch
+the OSProfile of the given instance, compare `existing_cves` there and update
+the `cves` field of the instance
+
+For Ubuntu, instead, the CVEs that affect an instance at runtime could be
+different from the `existing_cves` of the OSProfile, because new packages can be
+installed or upgraded at runtime. For this reason, we need a mechanism to report
+the CVEs that affect the runtime software. The Ubuntu OVAL data together with
+Security Content Automation Protocol (SCAP) compliant tools can be used to scan
+and Ubuntu system. This can be implemented in the Platform Update Agent, that
```

Comment:
Is there any example online how we can achieve that?

Reply:
Yes, here: https://ubuntu.com/security/oval

Reply:
Added reference